src/test/java/dk/cngroup/lentils/controller/AbstractClientControllerTest.java

```diff
@@ -0,0 +1,107 @@
+package dk.cngroup.lentils.controller;
+
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.CypherStatus;
+import dk.cngroup.lentils.entity.FinalPlace;
+import dk.cngroup.lentils.entity.Status;
+import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.entity.User;
+import dk.cngroup.lentils.repository.CypherRepository;
+import dk.cngroup.lentils.repository.FinalPlaceRepository;
+import dk.cngroup.lentils.repository.HintRepository;
+import dk.cngroup.lentils.repository.StatusRepository;
+import dk.cngroup.lentils.repository.TeamRepository;
+import dk.cngroup.lentils.repository.UserRepository;
+import dk.cngroup.lentils.security.CustomUserDetails;
+import dk.cngroup.lentils.service.ObjectGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.LocalDateTime;
+
+abstract public class AbstractClientControllerTest {
+    protected static final String CORRECT_CODEWORD = "topgun";
+    protected static final String FALSE_CODEWORD = "firefly";
+
+    @Autowired
+    protected ClientController testedController;
+    @Autowired
+    protected CypherRepository cypherRepository;
+    @Autowired
+    protected TeamRepository teamRepository;
+    @Autowired
+    protected UserRepository userRepository;
+    @Autowired
+    protected StatusRepository statusRepository;
+    @Autowired
+    protected ObjectGenerator generator;
+    @Autowired
+    protected FinalPlaceRepository finalPlaceRepository;
+    @Autowired
+    protected HintRepository hintRepository;
+
+    private Team team;
+    private User user;
+    private Cypher cypher;
+
+    protected void setupSharedFixture() {
+        createTestTeamAndUser();
+        createTestCypher();
+    }
+
+    protected Team getFixtureTeam() {
+        return team;
+    }
+
+    protected User getFixtureUser() {
+        return user;
+    }
+
+    protected Cypher getFixtureCypher() {
+        return cypher;
+    }
+
+    protected CustomUserDetails getUserDetailsMock() {
+        return new CustomUserDetails(user);
+    }
+
+    protected void setThatGameHasNotEnded() {
+        createFinalPlace(LocalDateTime.now().plusHours(5));
+    }
+
+    protected void setThatGameHasAlreadyEnded() {
+        createFinalPlace(LocalDateTime.now().minusHours(2));
+    }
+
+    protected Cypher createTestCypher() {
+        cypher = generator.generateValidCypher();
+        cypher.setCodeword(CORRECT_CODEWORD);
+        cypher.setTrapCodeword(FALSE_CODEWORD);
+
+        return cypherRepository.save(cypher);
+    }
+
+    protected void setCypherStatusForTeam(final Cypher cypher, final Team team, final CypherStatus value) {
```

Comment:
Napadlo mě akorát, že by se tu mohly prohodit  parametry Team team a CypherStatus value, aby to šlo podle názvu metody :).

Reply:
Může být.

Reply:
Parametry prohozeny 👍